src/spyglass/utils/dj_mixin.py

```diff
@@ -133,58 +153,146 @@ def _delete_deps(self) -> list:
         return self._delete_dependencies
 
     @property
-    def _merge_del_func(self) -> callable:
-        """Callable: delete_downstream_merge function.
+    def _test_mode(self) -> bool:
+        """Return True if test mode is enabled."""
+        if not self._test_mode_cache:
+            from spyglass.settings import test_mode
+
+            self._test_mode_cache = test_mode
+        return self._test_mode_cache
 
-        Used to delay import of func until needed, avoiding circular imports.
+    @property
+    def _merge_tables(self) -> Dict[str, dj.FreeTable]:
+        """Dict of merge tables downstream of self.
+
+        Cache of items in parents of self.descendants(as_objects=True) that
+        have a merge primary key.
         """
-        if not self._merge_delete_func:
-            from spyglass.utils.dj_merge_tables import (  # noqa F401
-                delete_downstream_merge,
-            )
+        if self._merge_table_cache:
+            return self._merge_table_cache
+
+        def has_merge_pk(table):
+            return MERGE_PK in table.heading.names
+
+        self.connection.dependencies.load()
+        for desc in self.descendants(as_objects=True):
+            if not has_merge_pk(desc):
+                continue
+            if not (master_name := get_master(desc.full_table_name)):
```

Comment:
![image](https://github.com/LorenFrankLab/spyglass/assets/8053989/de7b268d-e435-496c-87b8-df01736b08c9)

This is only available for Python 3.8 and above right? I suppose we are pinned above that.

Reply:
Yes. It's a novel operator for this codebase, so arguably less transparent to folks with less python experience, but my effort to keep the line count down in the mixin script